lms/djangoapps/instructor/views/api.py

```diff
@@ -433,9 +504,26 @@ def register_and_enroll_students(request, course_id):  # pylint: disable=too-man
                     # will raise an IntegrityError exception.
                     password = generate_unique_password(generated_passwords)
                     errors = create_and_enroll_user(
-                        email, username, name, country, password, course_id, course_mode, request.user, email_params
+                        email, username, name, country, password,
+                        course_id, course_mode,
+                        request.user,
+                        email_params, email_user=notify_by_email,
                     )
                     row_errors.extend(errors)
+                    if cohort:
+                        try:
+                            add_user_to_cohort(cohort, email)
+                        except ValueError:
+                            # user already in this cohort; ignore
+                            # NOTE: Checking this here may be unnecessary if we can prove that a new user will never be
+                            # automatically assigned to a cohort from the above.
+                            pass
+                        except ValidationError:
+                            row_errors.append({
+                                'username': username,
+                                'email': email,
+                                'response': _('Invalid email {email_address}.').format(email_address=email),
+                            })
```

Comment:
We already have this piece of code under this:
```python
            try:
                validate_email(email)  # Raises ValidationError if invalid
            except ValidationError:
                row_errors.append({
                    'username': username,
                    'email': email,
                    'response': _('Invalid email {email_address}.').format(email_address=email)
                })
            else:
                # … (more nestings)
                # This.
```
At this point, there should be no way to get this `ValidationError`.